plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java

```diff
@@ -185,19 +220,50 @@ private static org.apache.iceberg.types.Type fromRow(RowType type)
         for (RowType.Field field : type.getFields()) {
             String name = field.getName().orElseThrow(() ->
                     new TrinoException(NOT_SUPPORTED, "Row type field does not have a name: " + type.getDisplayName()));
-            fields.add(Types.NestedField.optional(fields.size() + 1, name, toIcebergType(field.getType())));
+            fields.add(Types.NestedField.optional(fields.size() + 1, name, toIcebergTypeInternal(type, Optional.empty())));
+        }
+        return Types.StructType.of(fields);
+    }
+
+    private static org.apache.iceberg.types.Type fromRow(RowType type, ColumnIdentity columnIdentity)
+    {
+        validateColumnIdentity(type.getFields().size(), columnIdentity);
+        List<Types.NestedField> fields = new ArrayList<>();
+        List<ColumnIdentity> childColumnIdentities = columnIdentity.getChildren();
+
+        for (RowType.Field field : type.getFields()) {
+            ColumnIdentity childColumnIdentity = childColumnIdentities.get(fields.size());
+            String name = field.getName().orElseThrow(() ->
+                    new TrinoException(NOT_SUPPORTED, "Row type field does not have a name: " + type.getDisplayName()));
+            fields.add(Types.NestedField.optional(childColumnIdentity.getId(), name, toIcebergTypeInternal(type, Optional.empty())));
```

Comment:
why do we use 
```
toIcebergTypeInternal(type, Optional.empty())
```

here? shouldn't it propagate the ids recursively?

Reply:
My fault